Example/Example/ViewControllers/DemoViewController.swift

```diff
@@ -0,0 +1,134 @@
+import UIKit
+import SalemoveSDK
+
+class DemoViewController: UIViewController {
+    @IBOutlet weak var engagementButton: UIButton!
+    @IBOutlet weak var configurationButton: UIButton!
+
+    private var statusViewController: EngagementStatusViewController?
+
+    // MARK: System
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        Salemove.sharedInstance.pushHandler = {
+            push in
+
+            // Do any additional refresh/updates when a user interacts with the notification
+        }
+    }
+
+    // MARK: Initialisation
+
+    @IBAction func queueMessage(_ sender: Any) {
+        let interactor = EngagementStatusViewController.initStoryboardInstance()
+        interactor.chatType = .async
+        Salemove.sharedInstance.configure(interactor: interactor)
+
+        statusViewController = interactor
+        statusViewController?.cleanUpBlock = { [unowned self] in
+            self.statusViewController?.dismiss(animated: true, completion: nil)
+            self.statusViewController = nil
+        }
+
+        let inputController = UIAlertController(title: "Message", message: "Please specify", preferredStyle: .alert)
+        inputController.addTextField { textfield in
+            textfield.clearButtonMode = .whileEditing
+            textfield.borderStyle = .roundedRect
+        }
+
+        let confirm = UIAlertAction(title: "Send", style: .default) { [unowned self] _ in
+            if let message = inputController.textFields?.first?.text, !message.isEmpty {
+                self.handleMessage(queueMessage: message)
+            }
+        }
+
+        let cancel = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+
+        inputController.addAction(confirm)
+        inputController.addAction(cancel)
+
+        present(inputController, animated: true, completion: nil)
+    }
+
+    @IBAction func beginQueueing(_ sender: Any) {
+        Salemove.sharedInstance.listQueues { [unowned self] queues, error in
+            if let queueError = error {
+                self.showError(message: queueError.reason)
+            } else if let queues = queues {
+                self.handleQueues(queues: queues)
```

Comment:
Let's rename `handleQueues` to something more specific. Like `showQueueSelection` or similar.